### PR TITLE
feat: support vite 3

### PR DESCRIPTION
**Pull request by sheremet-va**
Add support for Vite 3

TODO
- [ ] Updated dependencies to stable versions, when Vite 3 releases

### CODE REVIEW

File: examples/solid/vite.config.mjs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /// <reference types="vite/client" />
 
 import { defineConfig } from 'vite'
-import solid from 'solid-start' // or use 'vite-plugin-solid' instead
+import solid from 'vite-plugin-solid' // or use 'vite-plugin-solid' instead
```
Comment (sheremet-va) [inline examples/solid/vite.config.mjs:5]: We can remove comment now

File: packages/vitest/src/node/config.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ const extraInlineDeps = [
   /@vitest\/dist/,
   // Nuxt
   '@nuxt/test-utils',
+  /vite\w*\/dist\/client\/env.mjs/,
```
Comment (sheremet-va) [inline packages/vitest/src/node/config.ts:23]: This is backported to #1546

**Comment by sxzz** [comment]
It seems to take some time to wait for the stable release of Vite 3. Can we just release a major beta version of Vitest? Then we can use Vite 3 in Vitest.

Since Vite 2.x doesn't accept feature PR https://github.com/vitejs/vite/pull/8769 now.

**Comment by sheremet-va** [comment]
> It seems to take some time to wait for the stable release of Vite 3. Can we just release a major beta version of Vitest? Then we can use Vite 3 in Vitest.
> 
> Since Vite 2.x doesn't accept feature PR [vitejs/vite#8769](https://github.com/vitejs/vite/pull/8769) now.

We already support Vite 3, this is PR for updating mostly examples

**Comment by sxzz** [comment]
Oh, I see. So can we merge this PR #1540 now?

**Comment by sheremet-va** [comment]
> Oh, I see. So can we merge this PR #1540 now?

Sure

**Pull request closed without merging**